File: Linear/Static/Array/02_Inserting/End/Custom/Inserting.py
def insertAtEnd(arr, x):
    arr.append(0)

    # Insert x at the end.
    arr[-1] = x

    return arr

arr = [10, 20, 30, 40]
x = 50
print("Array before insertion")
for i in range(len(arr)):
    print(arr[i], end=" ")

insertAtEnd(arr, x)

print("\nArray after insertion")
for i in range(len(arr)):
    print(arr[i], end=" ")

File: Linear/Static/Array/02_Inserting/End/Custom/Inserting.java
class InsertingEndCustom {
    public static int[] insertAtEnd(int[] arr, int x) {
        int[] newArr = new int[arr.length + 1];
        for (int i = 0; i < arr.length; i++) {
            newArr[i] = arr[i];
        }

        // Insert x at the end.
        newArr[arr.length] = x;

        return newArr;
    }
    
    public static void main(String[] args) {
        int[] arr = {10, 20, 30, 40};
        int x = 50;
        System.out.println("Array before insertion");
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }

        int[] newArr = insertAtEnd(arr, x);

        System.out.println("\nArray after insertion");
        for (int i = 0; i < newArr.length; i++) {
            System.out.print(newArr[i] + " ");
        }
    }
}